Sources/tart/Serial.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  File.swift
+//  
+//
+//  Created by Peter Nguyen on 19/12/2022.
+//
+
+import Foundation
+
+func createPTY() -> Int32
+{
+    var tty_fd: Int32 = -1
+    var sfd: Int32 = -1
+    var termios_ = termios()
+    var tty_path = UnsafeMutablePointer<CChar>.allocate(capacity: 1024)
+    
+    let tty_fd_ref = UnsafeMutablePointer<Int32>.init(&tty_fd)
+    let sfd_ref = UnsafeMutablePointer<Int32>.init(&sfd)
+    let termios_ref = UnsafeMutablePointer<termios>.init(&termios_)
+    let unused_ref = UnsafeMutablePointer<winsize>.init(nil)
+    
+    var res = openpty(tty_fd_ref, sfd_ref, tty_path, termios_ref, unused_ref);
```

Comment:
Two things here:

1. Should we pass `nil` instead of `termios_ref` here to avoid using zeroed `termios` structure for the initial pseudo-TTY configuration? Then, call `tcgetattr()` before `cfmakeraw()` to actually populate the `termios` structure before modifying it.

2. Can we just pass `nil` here instead of `unused_ref`, and do not declare `unused_ref` at all?

Reply:
hi @edigaryev thank you for your reply I will fix these issues soon since I haven't familiar with swift before

Reply:
hi @edigaryev I have push some commits to clean code and add --serial option to tart

Reply:
How about we remove serial port from the configuration then? See https://github.com/cirruslabs/tart/pull/364#issuecomment-1362813600.

Reply:
Also, what do you think about item number 1 in https://github.com/cirruslabs/tart/pull/364#discussion_r1055435403?

Reply:
@edigaryev I agree with you to remove serial option in configuration file and I also removed it for the #1 I also cleared cfmakeraw because I didn't relate to tty setup.